pkg/conditions/upgrading.go

```diff
@@ -3,12 +3,67 @@ package conditions
 import (
 	aeconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions"
 	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
 	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
 	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
 
+	"github.com/giantswarm/azure-operator/v5/pkg/label"
 	"github.com/giantswarm/azure-operator/v5/service/controller/key"
 )
 
+func IsUpgradingInProgress(cr CR, desiredRelease string) (bool, error) {
+	c := capiconditions.Get(cr, aeconditions.UpgradingCondition)
```

Comment:
`Get` can return `nil`, so this can panic. There is a `capiconditions.IsTrue(cr, aeconditions.UpgradingCondition)` upstream func which checks that the condition is set and that it is set to `True`.